extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/graal/URLResourceSubstitution.java

```diff
@@ -73,7 +75,7 @@ private void openConnection() {
             URLClassLoader ucl = (URLClassLoader) cl;
             for (URL res : ucl.getURLs()) {
                 if (res.getProtocol().equals("file") && res.getPath().endsWith(".jar")) {
-                    try (JarFile file = new JarFile(res.getPath())) {
+                    try (JarFile file = new JarFile(URLDecoder.decode(res.getPath(), StandardCharsets.UTF_8.name()))) {
```

Comment:
I wonder if it would maybe be better to use `res.toUri().getPath()` instead?

Reply:
Readability wise yes :-) I'll change it.